frontend/src/modules/editor/themeEditor/views/editorThemingView.js

```diff
@@ -0,0 +1,576 @@
+      var selectedThemeId = selectedTheme.get('_id');
+      $.post('/api/theme/' + selectedThemeId + '/makeitso/' + this.model.get('_courseId'))
+        .error(this.onSaveError.bind(this))
+        .done(callback.bind(this));
+    },
+
+    postPresetData: function(callback) {
+      var selectedPreset = this.getSelectedPreset(false);
+      var selectedPresetId = null;
+      if (selectedPreset) selectedPresetId = selectedPreset.get('_id');
+
+      $.post('/api/themepreset/' + selectedPresetId + '/makeitso/' + this.model.get('_courseId'))
+      .error(this.onSaveError.bind(this))
+      .done(callback.bind(this));
+    },
+
+    postSettingsData: function(callback) {
+      if (!this.form) {
+        return callback.apply(this);
+      }
+      this.form.commit();
+      var settings = this.extractData(this.form.model.attributes);
+      Origin.editor.data.course.set('themeVariables', settings);
+      Origin.editor.data.course.save(null, {
+        error: this.onSaveError.bind(this),
+        success: callback.bind(this)
+      });
+    },
+
+    extractData: function(attributes) {
+      var data = {};
+      var properties = attributes.properties;
+      for (var key in properties) {
+        // Check for nested properties
+        if (typeof properties[key].properties !== 'undefined') {
+            data[key] = {};
+            for (var innerKey in properties[key].properties) {
+            data[key][innerKey] = attributes[innerKey];
+          }
+        } else {
+          data[key] = attributes[key];
+        }
+      }
+      return data;
+    },
+
+    navigateBack: function(event) {
+      event && event.preventDefault();
+      Backbone.history.history.back();
+      this.remove();
+    },
+
+    isDataLoaded: function() {
+      return this.themes.ready === true && this.presets.ready === true;
+    },
+
+    getSelectedTheme: function() {
+      var theme = $('select#theme', this.$el).val();
+      if (theme) {
+        return this.themes.findWhere({ 'theme': theme });
+      }
+      return this.themes.findWhere({ 'name': this.model.get('_theme') });
+    },
+
+    // param used to only return the val() (and ignore model data)
+    getSelectedPreset: function(includeCached) {
+      var storedId = this.getPresetSelection();
+      var presetId = $('select#preset', this.$el).val();
+      if (storedId) {
+        return this.presets.findWhere({ '_id': storedId });
+      }
+      if (presetId) {
+        return this.presets.findWhere({ '_id': presetId });
+      }
+      if (includeCached !== false){
+        var selectedTheme = this.getSelectedTheme();
+        if (!selectedTheme) return;
+        var parent = selectedTheme.get('theme');
+        return this.presets.findWhere({ '_id': this.model.get('_themePreset'), parentTheme: parent });
+      }
+    },
+
+    getDefaultThemeSettings: function() {
+      var defaults = {};
+      var props = this.getSelectedTheme().attributes.properties;
```

Comment:
I will look into this.

Reply:
I think within the variables object it is still going to be important to specify a title for each group of variables (colours, fonts etc.), so I think it will have to be more like this:

```
"variables": {
      "_colors": {
        "type": "object",
        "required": false,
        "title": "Colours",
        "properties": {
          "primary-color": {
            "title": "Primary",
            "help": "This is the primary colour used to style the course, and is applied to component items and pop-up notifications.",
            "type": "string",
            "inputType": "ColourPicker",
            "default": "#078292"
          }
        }
      }
    }
```

Reply:
That's what I was going for; just cut a bit too many attributes out of my snippet! Basically the same structure at present (nested & unnested) with a parent `variables` property.